src/ansys/mechanical/core/embedding/app.py

```diff
@@ -2,6 +2,9 @@
 import atexit
 import os
 import typing
+import warnings
+
+from pip._internal.operations import freeze
```

Comment:
I'm not a big fan of using "private" modules, but it's fine. However it looks like at older versions of pip this module is located in a different location:
```suggestion
try:
    from pip._internal.operations import freeze
except ImportError: # pip < 10.0
    from pip.operations import freeze
```